### PR TITLE
Make Ciao images immutable

### PR DESCRIPTION
**Pull request by rbradford**

### CODE REVIEW

**Comment by kaccardi** [comment]
@rbradford does the image service upload api need to be modified as well to enforce immutability? And should the ceph driver be modified to make images read only?

**Comment by rbradford** [comment]
@kaccardi Not sure if the image upload API can be made to enforce the immutability. Will check.

The ceph driver already makes rbd images used for images protected which will ensure that they cannot be modified. (They can be deleted if the reference count on them hits zero.)

**Comment by rbradford** [comment]
@kaccardi I've updated the patch to deny an override in the posix backend. The new ceph backend already does this.

File: ciao-image/datastore/posix.go [hunk 1]
```diff
@@ -29,6 +30,9 @@ type Posix struct {
 // If the image already exists it will be overridden.
 func (p *Posix) Write(ID string, body io.Reader) (int64, error) {
```
Comment (kaccardi) [inline ciao-image/datastore/posix.go]: can you change the comment?

**Comment by rbradford** [comment]
@kaccardi fixed.

**Pull request merged**